### DISCUSSION TITLE
Delayed actions correct setup - additional code not coding

### DISCUSSION THREAD

**Discussion by JohnySLO** [gopro/labs#429] [Q&A]
Hey guys! 

Hope that someone would have a suggestion on how to code this properly as unfortunately, I cannot find a correct combination in order to make things work. We will be using the gopro 9 on a remote location, where we won´t have access for a week and a half, therefore we would like it to work autonomously and remotely. 

The main points are:
it should perform a **Timelapse** with **Interval of 1 second**
it should start recording every 15 minutes for a period of 5 minutes (not to record a 5 minutes long timelapse as that would take much longer) - so 15 minutes break, 5 minutes recording, 15 minutes break etc.
The main point that I am not successful in is the limited period when the camera should work:
It should start recording every morning at 7AM and finish the recording everyday at 9PM and between these times record for 5 minutes after every 15 minutes as I mentioned previously. And repeat everyday like this.

Is there any way how we could possibly make it work?
Thank you for your help and time!!
Cheers Jan

**Comment by dnewman-gpsw** [comment]
Share what you have tried.  Command details, how the camera is powered, and the failure(s) you are experiencing.

**Reply by JohnySLO** [reply]
Hi!
So far we are in a trial version, the camera would be powered by battery and an additional power bank. We already managed to set up the correct period between the time lapses and the correct duration - however, we are still struggling with the border times between which it should record.

The main issues are:
the camera starts recording even before the set border time - so let say we want it to record after 07:00 AM but it happily works and records even at 6:45 for example.
It seems that it completely ignores the additional commands. - which are perhaps wrong.
So far we have tried such codes with no luck:
>!7:00S+<!21:00E!R
!07:00S+!21:00E
>07:00<21:00!S

GoPro QR: !900NmTPp1fW0r0!S!300E!R>07:00<21:00!S

**Reply by JohnySLO** [reply]
![image](https://github.com/gopro/labs/assets/138512632/73085f1f-f888-451d-ae2b-bb312599759a)

**Reply by dnewman-gpsw** [reply]
Unfortunately you have to choose between using the internal battery or external power, not both for these very long duration captures.  

This is backward, everything happens is order, so it will never decoding anything beyond !R (repeat)
 !900NmTPp1fW0r0!S!300E!R>07:00<21:00!S

Before I work to make a functional command, you need to consider the level of reliability you want. The most reliable solution is for the camera to remain on, and shooting continuously, day and night.  This I have personally done for 3 months without issues.  When sleeping the camera and setting wake alarms, that works pretty well with the internal battery, allowing interesting extended timelapses without any external equipment. When combining external power with the wake timer, reliability is not great. Reason (I think), the internal clock seems to forget time with USB Power alone, and alarms can fail to wake to camera. For a week long project, under external power, maybe 30% chance of success.  For very long timelapses you need an external timer, and running a very different script. 

What do you want to do for reliability?

**Reply by JohnySLO** [reply]
Thanks for the reply! We went through it with my supervisor now and we guess the external power supply would be the ideal solution, as we believe that the internal simply wouldn´t have sufficient capacity as we know that the GoPro battery tends to drain quite quickly, unfortunately.

thank you for your remark to the code! I´ll try to adjust it and see if it works :)

We were just thinking that the external battery would provide us with more reliability in terms of sufficient power, however, we would need to consider the size of the memory card if we would want to keep it running even during the night hours. Therefore I believe that high reliability would be our priority as these data are from a very remote location about 45 nautical miles from our island.

Thank you so much in advance!
